### PR TITLE
gucharmap: 3.18.2 -> 10.0.1

### PR DESCRIPTION
**Pull request by xzfc**
###### Motivation for this change

Update to the latest version.

- Gucharmap no longer follow the GNOME major versioning.
- Gucharmap no longer includes Unicode data. Instead the data should be downloaded from unicode.org.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/desktops/gnome-3/core/gucharmap/default.nix [hunk 1]
```diff
@@ -1,6 +1,7 @@
 { stdenv, intltool, fetchurl, pkgconfig, gtk3
 , glib, desktop_file_utils, bash, appdata-tools
-, makeWrapper, gnome3, file, itstool, libxml2 }:
+, wrapGAppsHook, gnome3, file, itstool, libxml2
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/core/gucharmap/default.nix]: `file` is no longer used.

Comment (xzfc) [reply pkgs/desktops/gnome-3/core/gucharmap/default.nix]: Removed `file`.

File: pkgs/desktops/gnome-3/core/gucharmap/src.nix [hunk 2]
```diff
@@ -1,10 +1,10 @@
 # Autogenerated by maintainers/scripts/gnome.sh update
```
Comment (orivej) [inline pkgs/desktops/gnome-3/core/gucharmap/src.nix]: This comment is obsolete.

File: pkgs/desktops/gnome-3/core/gucharmap/default.nix [hunk 3]
```diff
@@ -12,16 +13,17 @@ stdenv.mkDerivation rec {
...
 
-  preFixup = ''
-    wrapProgram "$out/bin/gucharmap" \
-      --prefix XDG_DATA_DIRS : "${gnome3.gnome_themes_standard}/share:$out/share:$GSETTINGS_SCHEMAS_PATH"
-  '';
+  buildInputs = [ gtk3 intltool itstool glib appdata-tools
+                  gnome3.yelp_tools libxml2 desktop_file_utils
+                  gnome3.gsettings_desktop_schemas ];
+
+  unicode-data = callPackage ./unicode-data.nix {};
```
Comment (orivej) [inline pkgs/desktops/gnome-3/core/gucharmap/default.nix]: This package seems useful outside gucharmap. For example, https://github.com/NixOS/nixpkgs/blob/26033a43/pkgs/tools/misc/unicode/default.nix could use it. However, I don't mind if it remains private for now.

File: pkgs/desktops/gnome-3/core/gucharmap/src.nix (hunk 2 above)
Comment (xzfc) [reply pkgs/desktops/gnome-3/core/gucharmap/src.nix]: No.
It is still generated by `./maintainers/scripts/gnome.sh update gucharmap`.
Having `gucharmap` in `NO_GNOME_MAJOR` means that this package is not updated with the rest of GNOME by calling `./maintainers/scripts/gnome.sh update-all`.

File: pkgs/desktops/gnome-3/core/gucharmap/default.nix (hunk 3 above)
Comment (xzfc) [reply pkgs/desktops/gnome-3/core/gucharmap/default.nix]: Yea.
However, I'm not sure which files it should contain (different apps may require different ucd files) and should it be zipped or not.
For example, `gucharmap` requires `Unihan.zip` to be zipped, but the contents of `UCD.zip` (`Blocks.txt`, `UnicodeData.txt`, ...) should be not.

**Comment by orivej** [comment]
Thank you for the clarifications.

File: pkgs/desktops/gnome-3/core/gucharmap/default.nix (hunk 3 above)
Comment (jtojnar) [reply pkgs/desktops/gnome-3/core/gucharmap/default.nix]: Arch package might be used as an inspiration: https://www.archlinux.org/packages/extra/any/unicode-character-database/.

**Pull request merged**